backtrader/feeds/psqlfeed

```diff
@@ -0,0 +1,101 @@
+# !/usr/bin/env python
+# -*- coding: utf-8; py-indent-offset:4 -*-
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+from backtrader import DataBase, date2num
+import datetime as dt
+import psycopg2
+from urllib.parse import urlparse
+
+
+# Used for fetching prices, adhering to the backtest library Database API
+
+
+class PsqlDatabase(DataBase):
+    '''
+    Fetches data from PSQL, and wraps it into a Feed consumable by cerebro
+    takes url connection string in form of :
+        postgres://{user}:{pw}@{host}:{port [optional]}/{database}
+
+    assumes a table name conforming to a schema similar to the following:
+            ticker        char(5),
+            date          date,
+            high          numeric(10,4),
+            low           numeric(10,4),
+            open          numeric(10,4),
+            close         numeric(10,4),
+            volume        integer,
+            unique (ticker, date)
+    '''
+
+    params = (
+        ('url', None),
+        ('dataname', None),
+        ('table', None),
+        ('fromdate', None),
+        ('todate', None),
+        ('ticker', None),
+    )
+
+    def start(self):
+        try:
+            conn = self._connect_db()
+            cursor = conn.cursor()
+        except Exception as e:
+            print('Failed to establish connection to PSQL: {}'.format(e))
```

Comment:
Printing error messages in inner components is probably good for local debugging - but not for a general way of error handling.  Users may have a bunch of reasons not to allow printing ( running in service environment, having different logging setup,  using progress bars, etc ).   

Suggest to just raise an exception with a proper description instead.

Reply:
good point! i'm going to remove the try/catch altogether so exceptions just propagate.